src/server/game/Globals/AreaTriggerDataStore.cpp

```diff
@@ -409,6 +409,26 @@ void AreaTriggerDataStore::LoadAreaTriggerSpawns()
             spawn.scriptId = sObjectMgr->GetScriptId(fields[12].GetString());
             spawn.spawnGroupData = sObjectMgr->GetLegacySpawnGroup();
 
+            // If the trigger has a teleport action, use it for checking go back/map entrance locations
+            auto teleportActionItr = std::ranges::find_if(createProperties->Template->Actions, [](AreaTriggerAction const& action) { return action.ActionType == AREATRIGGER_ACTION_TELEPORT; });
```

Comment:
maybe we should also explicitly check if destination map is != areatrigger mapid in case there are areatriggers teleporting within the same mapid